### ISSUE TITLE
Redirect-loop after logging in

### ISSUE THREAD

**Issue by ebogaard** [nextcloud/user_saml#22]
After enabling user_saml and configuring it to use out simplesamlphp-Idp, nextcloud redirects me nicely to the saml login screen. After succesfully logging in, there is some kind of redirect-loop:
1. I open the service and are redirected tot he saml-app - GET https://test-nc.DOMAIN/index.php/apps/user_saml/saml/login?requesttoken=GTM9QgIv....
2. the AuthnRequest is sent to the idp - GET https://dev-login.DOMAIN/saml/saml2/idp/SSOService.php?SAMLRequest=nVNNj9owE...
3. I redirected tot he SAML login-screen and POST the right credentials.
4. The request is answered by a Response. -  POST https://test-nc.DOMAIN/index.php/apps/user_saml/saml/acs
5. The response seems to contain the right info, among others the email address which is used as the username.
6. Then I'm redirected to the root of Nextcloud.
7. Now I expect to be logged in be redirected to "/index.php/apps". But that doesn't happen. The browser gets in some kind of loop and is redirected to step 1, step 2, step 4, then step 1 again, and so on.

The main configuration options in nextcloud:

```
"user_saml","general-uid_mapping","urn:oasis:names:tc:SAML:1.1:nameid-format:emailAddress"
"user_saml","idp-entityId","https://dev-login.DOMAIN/saml/saml2/idp/metadata.php"
"user_saml","idp-singleLogoutService.url","https://dev-login.DOMAIN/saml/saml2/idp/SingleLogoutService.php"
"user_saml","idp-singleSignOnService.url","https://dev-login.DOMAIN/saml/saml2/idp/SSOService.php"
"user_saml","idp-x509cert","-----BEGIN CERTIFICATE-----
MIIDkDCCAn...
-----END CERTIFICATE-----"
"user_saml","installed_version","1.0.1"
```

Note: test-nc = nextcloud / dev-login = saml idp.

I'm thinking this might have something to do with issue #13, but this might be another problem.

**Comment by LukasReschke** [comment]
The mapping is a little bit of a misleading term. I guess you want to map to the email address? In Shibboleth terms that would be "urn:oid:1.3.6.1.4.1.5923.1.1.1.6", see also https://help.nextcloud.com/t/how-does-the-user-saml-app-work/1419/2?u=lukasreschke

Probably some UX refinement is necessary on the setup step here. Let me think of something, probably we should move it into a separate step and one has to pick from arguments provided by the IDP.

**Comment by ebogaard** [comment]
Tried your suggestion, but no difference.
When I look at the oid's, it actually more looks like I need to use "urn:oid:0.9.2342.19200300.100.1.3", as that maps to the attribute 'mail' (see: https://www.incommon.org/federation/attributesummary.html).
But that didn't change things either.

I'm more suspecting of something wrong at the acs, as that should accept actually 'any' value, as I explicitly **did not set** the option 'Only allow authentication if an account is existent on some other backend. (e.g. LDAP)', which, at least as I read it, should always auto-create any user that the idp finds valid.

**Comment by ebogaard** [comment]
FInally fixed this.
Seems to be a combination of:

1. Using the wrong attribute. It was just as simple as to use 'mail' instead of the OID (we use the mail-attribute in ldap as user login).
1. After this, the loop was gone. Then another problem was exposed: in the sp I configured the ACS-url including 'index.php', while I configured Nextcloud/httpd to use a 'clean' URL without this part. This led to the situation that Chrome were redirected and couldn't login, while Firefox and IE were able to login properly. Leaving the index.php out of the ACS-url fixed that.
